src/components/PrimaryPanes/index.js

```diff
@@ -118,15 +131,18 @@ class PrimaryPanes extends Component<Props, State> {
 
 const mapStateToProps = state => ({
   selectedTab: getSelectedPrimaryPaneTab(state),
-  sources: getSources(state),
-  sourceSearchOn: getActiveSearch(state) === "source"
+  sources: getRelativeSources(state),
+  sourceSearchOn: getActiveSearch(state) === "source",
+  getWorkerDisplayName: thread => getWorkerDisplayName(state, thread)
```

Comment:
we've passed selectors into components in the past a couple of times. Generally, we tend to regret it for a couple of reasons:

1. it is a new function every time which causes react to try re-rendering pretty often
2. it tends to encourage us to put too much logic in the components

In this case, i think we can do two things:

1.  store an index in the debuggee/workers reducer field to represent the index that worker w/ the same URL appeared. 
2. query for the workers here and pass workers into a util `getWorkerDisplayName(workers, thread)` or something like that